File: src/components/JSONHandler/src/ALRPCObjectsImpl/V2/TireStatus.cpp
#include "../include/JSONHandler/ALRPCObjects/V2/TireStatus.h"
#include "TireStatusMarshaller.h"
#include "SingleTireStatusMarshaller.h"
#include "TirePressureTellTaleMarshaller.h"
/*
  interface	Ford Sync RAPI
  version	2.0O
  date		2012-11-02
  generated at	Tue Dec  4 14:30:13 2012
  source stamp	Tue Dec  4 14:21:32 2012
  author	robok0der
*/



using namespace NsAppLinkRPCV2;

TireStatus::TireStatus(const TireStatus& c)
{
  *this=c;
}


bool TireStatus::checkIntegrity(void)
{
  return TireStatusMarshaller::checkIntegrity(*this);
}


TireStatus::TireStatus(void)
{
}



bool TireStatus::set_innerLeftRear(const SingleTireStatus& innerLeftRear_)
{
  if(!SingleTireStatusMarshaller::checkIntegrityConst(innerLeftRear_))   return false;
  innerLeftRear=innerLeftRear_;
  return true;
}

bool TireStatus::set_innerRightRear(const SingleTireStatus& innerRightRear_)
{
  if(!SingleTireStatusMarshaller::checkIntegrityConst(innerRightRear_))   return false;
  innerRightRear=innerRightRear_;
  return true;
}

bool TireStatus::set_leftFront(const SingleTireStatus& leftFront_)
{
  if(!SingleTireStatusMarshaller::checkIntegrityConst(leftFront_))   return false;
  leftFront=leftFront_;
  return true;
}

bool TireStatus::set_leftRear(const SingleTireStatus& leftRear_)
{
  if(!SingleTireStatusMarshaller::checkIntegrityConst(leftRear_))   return false;
  leftRear=leftRear_;
  return true;
}

bool TireStatus::set_pressureTellTale(const TirePressureTellTale& pressureTellTale_)
{
  if(!TirePressureTellTaleMarshaller::checkIntegrityConst(pressureTellTale_))   return false;
  pressureTellTale=pressureTellTale_;
  return true;
}

bool TireStatus::set_rightFront(const SingleTireStatus& rightFront_)
{
  if(!SingleTireStatusMarshaller::checkIntegrityConst(rightFront_))   return false;
  rightFront=rightFront_;
  return true;
}

bool TireStatus::set_rightRear(const SingleTireStatus& rightRear_)
{
  if(!SingleTireStatusMarshaller::checkIntegrityConst(rightRear_))   return false;
  rightRear=rightRear_;
  return true;
}




const SingleTireStatus& TireStatus::get_innerLeftRear(void) const 
{
  return innerLeftRear;
}


const SingleTireStatus& TireStatus::get_innerRightRear(void) const 
{
  return innerRightRear;
}


const SingleTireStatus& TireStatus::get_leftFront(void) const 
{
  return leftFront;
}


const SingleTireStatus& TireStatus::get_leftRear(void) const 
{
  return leftRear;
}


const TirePressureTellTale& TireStatus::get_pressureTellTale(void) const 
{
  return pressureTellTale;
}


const SingleTireStatus& TireStatus::get_rightFront(void) const 
{
  return rightFront;
}


const SingleTireStatus& TireStatus::get_rightRear(void) const 
{
  return rightRear;
}

File: mobile/android/SyncProxyAndroid/src/com/ford/syncV4/proxy/rpc/TireStatus.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;

import com.ford.syncV4.proxy.RPCStruct;
import com.ford.syncV4.proxy.constants.Names;
import com.ford.syncV4.proxy.rpc.enums.TirePressureTellTale;
import com.ford.syncV4.util.DebugTool;

public class TireStatus extends RPCStruct {

    public TireStatus() { }
    public TireStatus(Hashtable hash) {
        super(hash);
    }
    public void setPressureTellTale(TirePressureTellTale pressureTellTale) {
    	if (pressureTellTale != null) {
    		store.put(Names.pressureTellTale, pressureTellTale);
    	} else {
    		store.remove(Names.pressureTellTale);
    	}
    }
    public TirePressureTellTale getPressureTellTale() {
        Object obj = store.get(Names.pressureTellTale);
        if (obj instanceof TirePressureTellTale) {
            return (TirePressureTellTale) obj;
        } else if (obj instanceof String) {
        	TirePressureTellTale theCode = null;
            try {
                theCode = TirePressureTellTale.valueForString((String) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.pressureTellTale, e);
            }
            return theCode;
        }
        return null;
    }
    public void setLeftFront(SingleTireStatus leftFront) {
    	if (leftFront != null) {
    		store.put(Names.leftFront, leftFront);
    	} else {
    		store.remove(Names.leftFront);
    	}
    }
    public SingleTireStatus getLeftFront() {
    	Object obj = store.get(Names.leftFront);
        if (obj instanceof SingleTireStatus) {
            return (SingleTireStatus) obj;
        } else if (obj instanceof Hashtable) {
        	try {
        		return new SingleTireStatus((Hashtable) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.leftFront, e);
            }
        }
        return null;
    }
    public void setRightFront(SingleTireStatus rightFront) {
    	if (rightFront != null) {
    		store.put(Names.rightFront, rightFront);
    	} else {
    		store.remove(Names.rightFront);
    	}
    }
    public SingleTireStatus getRightFront() {
    	Object obj = store.get(Names.rightFront);
        if (obj instanceof SingleTireStatus) {
            return (SingleTireStatus) obj;
        } else if (obj instanceof Hashtable) {
        	try {
        		return new SingleTireStatus((Hashtable) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.rightFront, e);
            }
        }
        return null;
    }
    public void setLeftRear(SingleTireStatus leftRear) {
    	if (leftRear != null) {
    		store.put(Names.leftRear, leftRear);
    	} else {
    		store.remove(Names.leftRear);
    	}
    }
    public SingleTireStatus getLeftRear() {
    	Object obj = store.get(Names.leftRear);
        if (obj instanceof SingleTireStatus) {
            return (SingleTireStatus) obj;
        } else if (obj instanceof Hashtable) {
        	try {
        		return new SingleTireStatus((Hashtable) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.leftRear, e);
            }
        }
        return null;
    }
    public void setRightRear(SingleTireStatus rightRear) {
    	if (rightRear != null) {
    		store.put(Names.rightRear, rightRear);
    	} else {
    		store.remove(Names.rightRear);
    	}
    }
    public SingleTireStatus getRightRear() {
    	Object obj = store.get(Names.rightRear);
        if (obj instanceof SingleTireStatus) {
            return (SingleTireStatus) obj;
        } else if (obj instanceof Hashtable) {
        	try {
        		return new SingleTireStatus((Hashtable) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.rightRear, e);
            }
        }
        return null;
    }
    public void setInnerLeftRear(SingleTireStatus innerLeftRear) {
    	if (innerLeftRear != null) {
    		store.put(Names.innerLeftRear, innerLeftRear);
    	} else {
    		store.remove(Names.innerLeftRear);
    	}
    }
    public SingleTireStatus getInnerLeftRear() {
    	Object obj = store.get(Names.innerLeftRear);
        if (obj instanceof SingleTireStatus) {
            return (SingleTireStatus) obj;
        } else if (obj instanceof Hashtable) {
        	try {
        		return new SingleTireStatus((Hashtable) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.innerLeftRear, e);
            }
        }
        return null;
    }
    public void setInnerRightRear(SingleTireStatus innerRightRear) {
    	if (innerRightRear != null) {
    		store.put(Names.innerRightRear, innerRightRear);
    	} else {
    		store.remove(Names.innerRightRear);
    	}
    }
    public SingleTireStatus getInnerRightRear() {
    	Object obj = store.get(Names.innerRightRear);
        if (obj instanceof SingleTireStatus) {
            return (SingleTireStatus) obj;
        } else if (obj instanceof Hashtable) {
        	try {
        		return new SingleTireStatus((Hashtable) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.innerRightRear, e);
            }
        }
        return null;
    }
}
